### DISCUSSION TITLE
Examples for observation

### DISCUSSION THREAD

**Discussion by zzd1992** [ds4dm/ml4co-competition#99] [Q&A] [answered]
min cx
s.t. Ax <= b,  l<x<u

Could anyone provide a complete example of how to convert the original problem into this formation and obtain c A b l u?
We don't need to consider discrete variables.
I want the raw data in array format.

**Comment by gasse** [comment, marked as answer]
Hi @zzd1992 ,

I think you'll find what you want here:
https://github.com/ds4dm/ml4co-competition/discussions/18#discussioncomment-976851

Best,
Maxime

**Reply by zzd1992** [reply]
@gasse 

Thanks for your response.

I also want to get the value of objective (cx) for the current solution.

How can I obtain it in the observation function?

I read the document of PySCIPOpt. But I don't find the way.

**Reply by gasse** [reply]
Here are some pointers:

[`x = m.getBestSol()`](https://github.com/scipopt/PySCIPOpt/blob/0485fdb17851749a80ca6254d1e9cc01e44c2c95/src/pyscipopt/scip.pyx#L4247)

[`cx = m.getSolObjVal(x)`](https://github.com/scipopt/PySCIPOpt/blob/0485fdb17851749a80ca6254d1e9cc01e44c2c95/src/pyscipopt/scip.pyx#L4252)

Best,
Maxime
